eq-author-api/app.js

```diff
@@ -20,38 +20,45 @@ const pino = pinoMiddleware();
 const logger = createLogger(pino.logger);
 addErrorLoggingToSchema(schema, logger);
 
-const knex = require("./db");
-
-const context = { repositories: repositories(knex) };
-
-app.use(
-  "/graphql",
-  pino,
-  cors(),
-  bodyParser.json(),
-  graphqlExpress({
-    schema,
-    context,
-    formatError: logger.log
-  })
-);
-
-app.get("/status", status);
-
-app.get("/launch/:questionnaireId", getLaunchUrl(context));
-if (process.env.NODE_ENV === "development") {
-  app.post("/import", bodyParser.json({ limit: "50mb" }), importAction);
-}
-
-if (process.env.NODE_ENV === "development") {
+const db = require("./db");
+
+db(process.env.DB_SECRET_ID).then(async conf => {
+  let knex = require("knex")(conf);
+  logger.log("Running Migrate");
+  await knex.migrate.latest();
+  logger.log("Ran Migrate");
+
+  const context = { repositories: repositories(knex) };
+
   app.use(
-    "/graphiql",
+    "/graphql",
     pino,
     cors(),
-    graphiqlExpress({ endpointURL: "/graphql" })
+    bodyParser.json(),
+    graphqlExpress({
+      schema,
+      context,
+      formatError: logger.log
+    })
   );
-}
 
-app.listen(PORT, "0.0.0.0", () => {
-  console.log(colors.green("Listening on port"), PORT);
+  app.get("/status", status);
+
+  app.get("/launch/:questionnaireId", getLaunchUrl(context));
+  if (process.env.NODE_ENV === "development") {
+    app.post("/import", bodyParser.json({ limit: "50mb" }), importAction);
+  }
+
+  if (process.env.NODE_ENV === "development") {
+    app.use(
+      "/graphiql",
+      pino,
+      cors(),
+      graphiqlExpress({ endpointURL: "/graphql" })
+    );
+  }
+
+  app.listen(PORT, "0.0.0.0", () => {
+    console.log(colors.green("Listening on port"), PORT);
+  });
 });
```

Comment:
I think we should add a catch to and exit the process with the error otherwise IIRC they will end as unhandledRejections which currently don’t crash the process